Crash in stdlib builder

```
Use --sandbox_debug to see verbose messages from the sandbox
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x52fe05]

goroutine 1 [running]:
main.replicateTree(0xc4200ca000, 0x99, 0xc4200cc000, 0xd4, 0xc42007c690, 0x660fc0, 0x30)
        external/io_bazel_rules_go/go/tools/builders/replicate.go:139 +0x1c5
main.replicate(0xc4200c8000, 0x95, 0xc4200840d0, 0xd0, 0xc420049ef8, 0x1, 0x1, 0xc4200104a0, 0x8)
        external/io_bazel_rules_go/go/tools/builders/replicate.go:162 +0x21f
main.run(0xc42000f010, 0x3b, 0x3f, 0x0, 0xc420010480)
        external/io_bazel_rules_go/go/tools/builders/stdlib.go:55 +0x2a8
main.main()
        external/io_bazel_rules_go/go/tools/builders/stdlib.go:75 +0x66
Target //cmd/kube-apiserver:kube-apiserver failed to build
INFO: Elapsed time: 368.058s, Critical Path: 14.29s
```

--- a/go/tools/builders/replicate.go
+++ b/go/tools/builders/replicate.go
@@ -129,14 +129,14 @@ func replicateTree(src, dst string, config *replicateConfig) error {
 	if err := os.RemoveAll(dst); err != nil {
 		return fmt.Errorf("Failed to remove file at destination %s: %v", dst, err)
 	}
-	if l, err := filepath.EvalSymlinks(src); err == nil {
+	if l, err := filepath.EvalSymlinks(src); err != nil {
+		return err
+	} else {
 		src = l
 	}
-	s, err := os.Stat(src)
-	if err != nil && !os.IsNotExist(err) {
+	if s, err := os.Stat(src); err != nil {
 		return err
-	}
-	if s.IsDir() {
+	} else if s.IsDir() {
 		return replicateDir(src, dst, config)
 	}
 	return replicateFile(src, dst, config)

